awswrangler/athena/_read.py

```diff
@@ -736,6 +748,7 @@ def read_sql_query(  # pylint: disable=too-many-arguments,too-many-locals
     data_source: Optional[str] = None,
     athena_query_wait_polling_delay: float = _QUERY_WAIT_POLLING_DELAY,
     params: Optional[Dict[str, Any]] = None,
+    execution_params: Optional[List[str]] = None,
```

Comment:
I'm not a big fan of having these in addition to `params`. However, these Athena parameterized queries are a key part of prepared statements.

They also aren't as flexible as the current setup with `params`, which automatically formats strings.

Reply:
It's not just for `EXECUTE` statements though. You can also have a query like:
```sql
SELECT * FROM table WHERE col = ?
```
`params` uses client-side parameter resolution, meaning that the SQL query sent in Athena includes all the parameters in the query itself. The major downside is that this won't be compatible with executing prepared statements.

However, it the situations where it works, it works a lot better than `execution_params`. Parameters are named, meaning that the order in which you define parameters doesn't matter. Secondly, it also automatically resolves types. Instead of needing to pass `DECIMAL '2.1'`, you can pass a Python object of type `Decimal`. The `execution_params` parameter doesn't offer that.

I agree with Abdel that a tutorial is necessary to illustrate the difference. But I don't see a way where we can consolidate these two into a single parameter.